lms/djangoapps/course_home_api/outline/v1/views.py

```diff
@@ -99,6 +113,13 @@ def get(self, request, *args, **kwargs):
         show_handouts = is_enrolled or is_staff or allow_public
         handouts_html = get_course_info_section(request, request.user, course, 'handouts') if show_handouts else ''
 
+        welcome_message_html = None
+        if get_course_tag(request.user, course_key, PREFERENCE_KEY) != 'False':
```

Comment:
This can just be `if get_course_tag...`

Reply:
Also, what is this? Looking at course_home.py, the only check that happens before getting the HTML is 
```
if user_access['is_enrolled'] or user_access['is_staff']:
```
I do see this `get_course_tag` show up inside the WelcomeMesssageFragmentView, but it is only there so it's odd that we are blocking both on that condition.

Reply:
if get_course_tag(request.user, course_key, PREFERENCE_KEY) tells us if the user has disabled the welcome message.